helix/appendixes/package-principles.rst

```diff
@@ -0,0 +1,131 @@
+audience, such as developers working on navigation. This principle lets them
+treat a module as a well-defined space in which only classes relevant to their
+development effort are present, limiting distraction, complexity and the 
+potential for unintended side effects. In Helix terms, this principle could
+be restated as, "The granule of business purpose is the granule of module size,"
+or more simply, "A module should serve a single business audience."
+
+The Common Reuse Principle (CRP)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    The classes in a component are reused together. If you reuse one of the
+    classes in a component, you reuse them all.
+
+In Martin's writing, this principle establishes that there are close relationships
+between the classes in a component: "classes that are not bound to each other with 
+class relationships should not be in the same component."  We see this principle at work
+in the treatment of items and templates in Helix. By defining items through the
+composition of interface templates, Helix hides from the module the fields the 
+module does not use. Developers familiar with Martin's SOLID principles will
+recognize this as an application of the Interface Segregation Principle: "Clients 
+should not be forced to depend on methods they do not use" (Martin, ch. 12).
+
+By using interface templates, and limiting item visibility to these fields, Helix 
+keeps out of the view of the module functionality of other modules, as well as all the
+fields Sitecore uses to manage items (icon, publishing settings, workflow, 
+security). This is why even with Data Source components, Helix recommends a separation
+between the interface template, owned by the module, and the Data Source template, 
+owned by the Project layer (see `Template types`_).
+
+.. _Template types: principles/templates/template-types.html
+ 
+The Common Closure Principle (CCP)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    The classes in a component should be closed together against the same 
+    kinds of changes. A change that affects a component affects all the 
+    classes in that component and no other components.
+
+This principle seeks to limit the impact of changes to one or a small set of 
+modules, so that changes do not ripple across all the components of a system. 
+This principle motivates the Helix practice of grouping different types of elements
+(templates, renderings, views, controllers) that collaborate to produce a feature.
+In this way, a change to that feature will land entirely within the scope of one
+module. For example, imagine adding a new behavior to a navigation component, 
+with a new field added to an interface template, new code added to a model repository,
+and new markup added to CSHTML. Although these are different technologies, they are 
+bound in one Helix module, and thus the impact of the modification is limited to 
+that module. 
+
+Principles of Component Coupling
```

Comment:
Same as above, isn't the correct name "Principles of Package Coupling"?

Reply:
It's components, at least in the C# edition.

![image](https://user-images.githubusercontent.com/689532/67248042-19817480-f431-11e9-895c-f2cd5e0dac0e.png)

Reply:
As above, we should be consistent with out naming, maybe introduce content explaining package vs component?

Reply:
Text added:

> A note on terms: Martin has written about these principles in a number of places, and sometimes uses the term "package" and
sometimes "component." For consistency, we will use "package" throughout this appendix when referring to Martin's writing,
and "module" when referring to the Helix unit.